src/test/java/crawlercommons/robots/SimpleRobotRulesParserTest.java

```diff
@@ -713,6 +714,30 @@ void testManyUserAgents() throws Exception {
         assertFalse(rules.isAllowed("http://domain.com/bot-trap/"), "many-user-agents");
     }
 
+    @Test
+    void testManyUserAgentsMergeRules() throws Exception {
+        BaseRobotRules rules = createRobotRules("wget", readFile("/robots/merge-rules.txt"));
+        assertTrue(rules.isAllowed("http://domain.com/foo/bar/bar.txt"), "merge-rules");
+        assertFalse(rules.isAllowed("http://domain.com/tmp/bar/bar.txt"), "merge-rules");
+        assertTrue(rules.isAllowed("http://domain.com/chk/bar/bar.txt"), "merge-rules");
+    }
+
+    @Test
+    void testConflictingRules() throws Exception {
+        BaseRobotRules rules = createRobotRules("wget", readFile("/robots/conflicting-rules.txt"));
+
+        assertTrue(rules.isAllowed("http://domain.com/"), "conflicting-rules");
+        assertFalse(rules.isAllowed("http://domain.com/foo/bar"), "conflicting-rules");
```

Comment:
See comment about `/*` in `/robots/conflicting-rules.txt`: If `/*` is equivalent to `/` than `/foo/bar` would be allowed?

Reply:
Correct, if we go by Google's documentation. Their implementation however, does not. 

One thing I found through addressing your comments is that google applies special rules to Rules such as:

```
Allow: /index.html
```

They actually process this twice, once for an exact match, and a second time where they handle it as
```
Allow: /$
```

This is what allows robots.txt such as https://venmo.com/robots.txt to work as intended (through not by the spec). I've adjusted this PR to take this into account as well.